### PR TITLE
WIP: Allow using an HTTP header for auth

### PR DESCRIPTION
**Pull request by pcarranzav**
connects to #231

---
This change is [Reviewable](https://reviewable.io/reviews/resin-io/resin-supervisor/232)

### CODE REVIEW

**Comment by pcarranzav** [comment]
@petrosagg @Page-

**Comment by petrosagg** [comment]
SGTM

File: src/api.coffee [hunk 1]
```diff
@@ -17,9 +17,15 @@ module.exports = (application) ->
 	parsedRouter.use(bodyParser())
 
 	api.use (req, res, next) ->
+		queryKey = req.query.apikey
+		header = req.get('Authorization')
+		match = header.match(/^ApiKey ([a-f\d]{64})$/)
```
Comment (Page-) [inline src/api.coffee]: I don't think it makes sense for this to check for a fixed length/characters, for instance it won't work with the dev api key of 'bananas'

Comment (pcarranzav) [inline src/api.coffee]: True, fixed

File: src/api.coffee [hunk 2]
```diff
@@ -17,9 +17,15 @@ module.exports = (application) ->
 	parsedRouter.use(bodyParser())
 
 	api.use (req, res, next) ->
+		queryKey = req.query.apikey
+		header = req.get('Authorization')
+		match = header.match(/^ApiKey ([a-f\d]+)$/)
```
Comment (Page-) [inline src/api.coffee]: This still won't work for 'bananas', those troublesome `n`s, not to mention that `s` :P

Comment (pcarranzav) [inline src/api.coffee]: :banana: damn, now really fixed

File: src/api.coffee [hunk 3]
```diff
@@ -17,9 +17,15 @@ module.exports = (application) ->
 	parsedRouter.use(bodyParser())
 
 	api.use (req, res, next) ->
+		queryKey = req.query.apikey ? ''
```
Comment (Page-) [inline src/api.coffee]: You check for `queryKey?` below, but this line sets a default so that one will never be nullish

Comment (pcarranzav) [inline src/api.coffee]: Today's clearly not my brightest day. Fixed.

**Comment by Page-** [comment]
LGTM

**Pull request merged**